tests/extproc/extproc_test.go

```diff
@@ -50,6 +50,16 @@ func TestE2E(t *testing.T) {
 	openAIAPIKey := getEnvVarOrSkip(t, "TEST_OPENAI_API_KEY")
 	requireRunEnvoy(t, accessLogPath, openAIAPIKey)
 	configPath := t.TempDir() + "/extproc-config.yaml"
+
+	// Test with APIKey.
+	apiKeyFilePath := t.TempDir() + "/open-ai-api-key"
+	file, err := os.Create(apiKeyFilePath)
+	require.NoError(t, err)
+	_, err = file.WriteString(openAIAPIKey)
+	require.NoError(t, err)
+	require.NoError(t, file.Sync())
+	require.NoError(t, file.Close())
```

Comment:
nit, but let's close inside defer. If the error happens before close, say in `file.Sync()` then there would be a dangling  opened file
```suggestion
	defer func() { require.NoError(t, file.Close()) }()
	_, err = file.WriteString(openAIAPIKey)
	require.NoError(t, err)
	require.NoError(t, file.Sync())
```